### PR TITLE
Update to `image-library` branch of `cesium-native`

### PR DESCRIPTION
**Pull request by j9liu**
## Description

Depends on https://github.com/CesiumGS/cesium-native/pull/1388 so merge that first.

This PR updates code for the new `CesiumImage` namespace in cesium-native.

## Author checklist

- [X] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [X] I have done a full self-review of my code.
- ~[ ] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).~
- [x] I have added or updated unit tests to ensure consistent code coverage as necessary.
- ~[ ] I have updated the documentation as necessary.~

## Testing plan

Spotcheck the Samples project for various features that make use of images. For example:

- [ ] `06_CesiumMetadata` -> uses feature ID textures
- [ ] `08_CesiumClipping` -> clipping raster overlay

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: Source/CesiumRuntime/Private/Cesium3DTileset.cpp [hunk 1]
```diff
@@ -44,6 +44,7 @@
 #include "UnrealPrepareRendererResources.h"
 #include "VecMath.h"
 
+THIRD_PARTY_INCLUDES_START
```
Comment (timoore) [inline Source/CesiumRuntime/Private/Cesium3DTileset.cpp:47]: This is the first time I've seen `THIRD_PARTY_INCLUDES_START` and _END, though I  now see it in some other places. Obviously an Unreal thing... does it need to be documented in our own developer documentation somewhere?

**Review by timoore — COMMENTED** [review]
Looks good @j9liu . I like the new forward declarations that have come in as part of your refactor.

File: Source/CesiumRuntime/Private/Cesium3DTileset.cpp (hunk 1 above)
Comment (j9liu) [reply Source/CesiumRuntime/Private/Cesium3DTileset.cpp:47]: It probably should, thanks for the callout. I'll make an issue.
